sql/localsql/nipost_state.go

```diff
@@ -136,6 +137,30 @@ func discardNipostChallenge(dir string) error {
 	return nil
 }
 
+func saveBuilderState(dir string, state *NIPostBuilderState) error {
+	if err := save(filepath.Join(dir, builderFilename), state); err != nil {
+		return fmt.Errorf("saving builder state: %w", err)
+	}
+	return nil
+}
```

Comment:
The `saveBuilderState` and `discardBuilderState` are only used in tests. How about moving them to _test.go?

Reply:
`saveBuilderState` is only used by the test and I can move it there, but `discard-` and `loadBuilderState` are needed by the migration.